docs/join-mainnet.md

```diff
@@ -0,0 +1,190 @@
+# Join the Mainnet or Public Testnet
+
+### Install Go
+
+Install `go` by following the [official docs](https://golang.org/doc/install). Remember to set your `$GOPATH` and `$PATH` environment variables, for example:
+
+```bash
+mkdir -p $HOME/go/bin
+echo "export GOPATH=$HOME/go" >> ~/.bash_profile
+echo "export PATH=\$PATH:\$GOPATH/bin" >> ~/.bash_profile
+source ~/.bash_profile
```

Comment:
You are assuming here that the user shell is bash. This may work on Mac OS X, though it would definitely not work with all shells and the configuration may turn out to be incorrect in some environments. I encourage you to investigate the functionalities and differences between the `.profile`, `.bash_profile`, and `.bashrc` files.

I'd rather drop a link pointing to https://golang.org/doc/install instead of making assumptions on users shell of choice.